### PR TITLE
B-03763 - handle signals

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

File: bin/chaperone-server [hunk 1]
```diff
@@ -1,123 +1,123 @@
 #!/usr/bin/env node
 
-const path				= require('path');
-const log				= require('@whi/stdlog')(path.basename( __filename ), {
-    level: process.env.LOG_LEVEL || 'fatal',
+const path = require('path');
```
Comment (robbiecarlton) [inline bin/chaperone-server:3]: All the changes in this file are formatting changes, except one which I'll point out in a comment below

File: tests/mock_envoy.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ const mock_envoy_server = function (api_methods, port = 4656) { // 4656 = Holo |
   server.port = port;
   const namespace = "/hosting/";
 
+  server.event('signal', namespace);
```
Comment (robbiecarlton) [inline tests/mock_envoy.js:24]: attn @peeech - you will probably have to do something like this and line 116 of this file for Envoy

File: bin/chaperone-server (hunk 1 above)
Comment (robbiecarlton) [inline bin/chaperone-server:3]: changes in this file are all formatting.

File: src/index.ts [hunk 3]
```diff
@@ -634,7 +636,7 @@ class Chaperone {
     if (this.opts.connection.host) {
       if (this.mode === Chaperone.HCC) {
         log('connecting to conductor app interface on port (%s) ', this.port);
-        this.appWebsocket = await AppWebsocket.connect(`ws://localhost:${this.port}`);
+        this.appWebsocket = await AppWebsocket.connect(`ws://localhost:${this.port}`, this.sendSignal.bind(this))
```
Comment (peeech) [inline src/index.ts:639]: second paramenter of `.connect()` is timeout.

**Review by peeech — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 4]
```diff
@@ -356,6 +357,10 @@ class Chaperone {
     }
   }
 
+  async sendSignal(signal) {
+    this.parent_window.sendSignal(signal)
```
Comment (peeech) [inline src/index.ts:361]:
```suggestion
    if (!this.parent_window)
        return;
    this.parent_window.sendSignal(signal)
```

Comment (peeech) [reply src/index.ts:361]: Accounts for a case when parent_window is not set up yet by `COMB.listen()`.

File: src/index.ts (hunk 3 above)
Comment (peeech) [reply src/index.ts:639]: actually not true at all. Sorry.

**Review by peeech — APPROVED** [review]

File: src/index.ts (hunk 3 above)
Comment (robbiecarlton) [reply src/index.ts:639]: It depends on the version of `holochain-conductor-api`. The api changed. The newer version does have timeout as its second param. But we're still on `0.0.1-dev.14`

**Pull request merged**